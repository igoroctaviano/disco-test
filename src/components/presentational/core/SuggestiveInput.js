import React from "react";
import PropTypes from 'prop-types';
import UnorderedList from "./UnorderedList.js";

const SuggestiveInput = ({ onChange, items, itemRenderer, value }) => {
  return (
    <div className="suggestive-input">
      <div className="wrap">
        <span className="icon" />
        <input autoFocus onChange={onChange} value={value} />
      </div>
      {items.length > 0 ? (
        <UnorderedList items={items} itemRenderer={itemRenderer} />
      ) : null}
    </div>
  );
};

SuggestiveInput.propTypes = {
  onChange: PropTypes.func,
  items: PropTypes.array,
  itemRenderer: PropTypes.func,
  value: PropTypes.string
};

export default SuggestiveInput;
