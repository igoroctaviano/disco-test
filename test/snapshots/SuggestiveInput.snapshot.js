import React from "react";
import SuggestiveInput from "./../../src/components/presentational/core/SuggestiveInput.js";
import renderer from "react-test-renderer";
import { items, itemRenderer } from "./mocks.js";

test("renders correctly", () => {
  const tree = renderer
    .create(<SuggestiveInput items={items} itemRenderer={itemRenderer} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
