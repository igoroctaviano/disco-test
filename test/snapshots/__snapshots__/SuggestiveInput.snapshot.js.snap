// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="suggestive-input"
>
  <div
    className="wrap"
  >
    <span
      className="icon"
    />
    <input
      autoFocus={true}
    />
  </div>
  <ul>
    1
    2
    3
  </ul>
</div>
`;
